staging/src/k8s.io/kubectl/pkg/validation/validation.go

```diff
@@ -27,19 +28,42 @@ import (
 	"k8s.io/kubectl/pkg/util/openapi"
 )
 
+// OpenAPIResourcesGetter represents a function to return
+// OpenAPI V2 resource specifications. Used for lazy-loading
+// these resource specifications.
+type OpenAPIResourcesGetter interface {
+	OpenAPISchema() (openapi.Resources, error)
+}
+
 // schemaValidation validates the object against an OpenAPI schema.
 type schemaValidation struct {
+	resourcesGetter OpenAPIResourcesGetter
+
+	// Cached OpenAPI V2 resources specs
```

Comment:
This should be hidden behind the getter, i.e. we should have a cachedgetter or not. We already have such things that get tne openapi v2 document behind caches, that's pretty much to do that so we shouldn't have to reimplement it here.

We also already have the `CachedOpenAPIParser` :
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/util/openapi/openapi_getter.go#L56-L63

This lacks an `interface` though, but I'm sure we can re-use that.